File: src/main/java/travelator/mobile/UserPreferences.java
package travelator.mobile;

import java.util.Currency;
import java.util.Locale;

public class UserPreferences {

    private String greeting;
    private Locale locale;
    private Currency currency;

    public UserPreferences() {
        this("Hello", Locale.UK, Currency.getInstance(Locale.UK));
    }

    public UserPreferences(String greeting, Locale locale, Currency currency) {
        this.greeting = greeting;
        this.locale = locale;
        this.currency = currency;
    }

    public String getGreeting() {
        return greeting;
    }

    public void setGreeting(String greeting) {
        this.greeting = greeting;
    }

    public Locale getLocale() {
        return locale;
    }

    public void setLocale(Locale locale) {
        this.locale = locale;
    }

    public Currency getCurrency() {
        return currency;
    }

    public void setCurrency(Currency currency) {
        this.currency = currency;
    }
}

File: src/main/java/travelator/mobile/UserPreferences.kt
package travelator.mobile

import java.util.*

class UserPreferences(
    var greeting: String,
    var locale: Locale,
    var currency: Currency
) {
    constructor() : this(
        greeting = "Hello",
        locale = Locale.UK,
        currency = Currency.getInstance(Locale.UK)
    )
}